### PR TITLE
Enable runtime compilation & kernel header downloading for USM

### PR DESCRIPTION
**Pull request by ISauve**
### What does this PR do?

Turns on runtime compilation and kernel header downloading by default if Universal Service Monitoring is turned on.
Either feature can explicitly disabled in the system-probe configuration file.

### Motivation

[RFC: Requiring runtime compilation for Universal Service Monitoring](https://docs.google.com/document/d/1BzlDqyGY7pdfeDnl1tXUz3ZzUi23a56IMyZoefhfOg4/edit)

### Describe how to test/QA your changes

1) Enable only NPM:
```
network_config:
  enabled: true
```
Run the system-probe and verify in the logs that runtime compilation doesn't occur.

2) Enable USM with runtime compilation explicitly disabled:
```
service_monitoring_config:
  enabled: true
system_probe_config:
  enable_runtime_compiler: false
network_config:
  enabled: true
```
Run the system-probe and verify in the logs that runtime compilation doesn't occur.

3) Finally, enable USM:
```
service_monitoring_config:
  enabled: true
network_config:
  enabled: true
```
Run the system-probe and verify in the logs that runtime compilation does occur.

### Reviewer's Checklist

- [x] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [x] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [x] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature. 
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

File: pkg/network/config/config.go
Comment (brycekahle) [inline pkg/network/config/config.go:277]: Do we also want to disable fallback, since we would be requiring runtime compilation?

Comment (brycekahle) [reply pkg/network/config/config.go:277]: Sounds like that decision is still up in the air.

**Comment by brycekahle** [comment]
I'm wondering if we need to force RC off in the tests, since we already test with RC explicitly turned on. That way we are still testing the prebuilt fallback in CI.

File: pkg/network/config/config.go
Comment (ISauve) [reply pkg/network/config/config.go:277]: We'll likely defer this to 7.40 and keep the fallback to offset guessing for this release; waiting on the RFC to be finished for the final decision though

**Review by brycekahle — APPROVED** [review]

**Pull request merged**
